guac_core/src/storages/in_memory.rs

```diff
@@ -0,0 +1,100 @@
+use channel_client::types::{Channel, ChannelStatus};
+use clarity::Address;
+use crypto::CryptoService;
+use failure::Error;
+use futures::future::ok;
+use futures::Future;
+use num256::Uint256;
+use payment_contract::ChannelId;
+use qutex::Qutex;
+use rand;
+use rand::prelude::*;
+use rand::RngCore;
+use std::collections::HashMap;
+use storage::Storage;
+use CRYPTO;
+
+struct ChannelData {
+    url: String,
+
+    // TODO: Channel structure needs some changes but we can reuse it at this point
+    channel: Channel,
+}
+
+/// A in-memory storage that stores data in
+pub struct InMemoryStorage {
+    channels: Qutex<HashMap<ChannelId, ChannelData>>,
+}
+
+impl InMemoryStorage {
+    pub fn new() -> Self {
+        Self {
+            channels: Qutex::new(HashMap::new()),
+        }
+    }
+}
+
+impl Storage for InMemoryStorage {
+    /// Registers new counterparty
+    ///
+    /// * `url` - Remote URL
+    /// * `url` - Remote ETH address
+    /// * `balance` - Our initial deposit
+    fn register(
+        &self,
+        url: String,
+        address: Address,
+        balance: Uint256,
+    ) -> Box<Future<Item = Channel, Error = Error>> {
+        // rand::thread_rng()
+        let channel_id: ChannelId = {
```

Comment:
You should be able to do this in one line with `rand::rand()` (your typedef for `ChannelId` should be infered here). `Random` is implemented for `[u8; <whatever>]`.

Reply:
I tried that but for some reason it wasn't working. Probably it was just unrelated error. Fixed